src/options.c

```diff
@@ -525,25 +525,19 @@ static void showVersion(void)
 
 char *optionsNameThumbnail(const char *name)
 {
+    const size_t nameLength = strlen(name);
     const char thumbSuffix[] = "-thumb";
-    const size_t newNameLength = strlen(name) + sizeof(thumbSuffix);
-    char *newName = malloc(newNameLength);
-
-    if (!newName)
-        err(EXIT_FAILURE, "Unable to allocate thumbnail");
-
+    Stream ret = {0};
     const char *const extension = strrchr(name, '.');
+    const size_t baseNameLength = extension ? extension-name : nameLength;
 
-    if (extension) {
-        /* We add one so length includes '\0'*/
-        const ptrdiff_t nameLength = (extension - name) + 1;
-        strlcpy(newName, name, nameLength);
-        strlcat(newName, thumbSuffix, newNameLength);
-        strlcat(newName, extension, newNameLength);
-    } else
-        snprintf(newName, newNameLength, "%s%s", name, thumbSuffix);
+    streamMem(&ret, name, baseNameLength);
+    streamMem(&ret, thumbSuffix, sizeof(thumbSuffix)-1);
+    if (extension)
```

Comment:
>don't pass NULL to streamMem
>
>This is actually perfectly valid but it looks weird.

The code is fine after this commit, but wanted to clarify that passing NULL to any of the string.h functions is UB _even if_ `n` is zero [^0] (and compilers like gcc/clang can aggressively optimize code around such calls with the assumptions that the pointer are non-null just because it was an argument in a memcpy call).

[^0]: https://port70.net/~nsz/c/c99/n1256.html#7.21.1p2

> Where an argument declared as size_t n specifies the length of the array for a function, n can have the value zero on a call to that function. Unless explicitly stated otherwise in the description of a particular function in this subclause, pointer arguments on such a call shall still have valid values